### PR TITLE
feat: Allow passing parameters in connection string

### PR DESCRIPTION
**Pull request by elefeint**
Currently the only way to pass settings to DuckDB is through the `connect_args` parameter. 

This PR allows URL parameters in the connection string [parsed by SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy/blob/14825668ab407115f793dd229ed694c5c607c657/lib/sqlalchemy/engine/create.py#L548) to get passed back to the framework from this driver's `create_connect_args()` function. I did something similar to SQLAlchemy's [reference implementation](https://github.com/sqlalchemy/sqlalchemy/blob/14825668ab407115f793dd229ed694c5c607c657/lib/sqlalchemy/engine/interfaces.py#L1233-L1236), but allowing for config mutability since it will get split into core and extended properties in `connect()` method.

Affiliation: MotherDuck

### CODE REVIEW

**Comment by Mause** [comment]
Hi @elefeint, can you explain why you want this? Are you using an application that doesn't allow you to specify connect_args?

Also, I'd appreciate it you could find the time to fix the various ci failures and warnings

**Comment by elefeint** [comment]
@Mause Sorry for the failures! 

The reason for the PR is that when troubleshooting something unrelated, I noticed that connection string parameters behaved differently than `connect_args` - they got ignored. This would pause no difficulty to users accessing DuckDB through SQLAlchemy programmatically, since they can use `connect_args`, but for users of 3rd party integrations built on top of SQLAlchemy such as [Superset](https://superset.apache.org/docs/databases/installing-database-drivers/), the connection string is often the only settable configuration the user has access to.

This isn't blocking anything I am currently working on, but it seemed like a nice harmless improvement... until my lack of Python knowledge caught up with me...

File: duckdb_engine/tests/test_basic.py [hunk 1]
```diff
@@ -363,6 +363,18 @@ def test_config(tmp_path: Path) -> None:
             conn.execute(text("create table hello2 (i int)"))
 
 
+def test_url_config(tmp_path: Path) -> None:
+    eng = create_engine("duckdb:///:memory:?worker_threads=123")
+
+    with eng.connect() as conn:
+        res = conn.execute(
+            text("select value from duckdb_settings() where name='worker_threads'")
```
Comment (Mause) [inline duckdb_engine/tests/test_basic.py:371]:
```suggestion
            text("select current_setting('worker_threads')")
```

File: duckdb_engine/tests/test_basic.py [hunk 2]
```diff
@@ -363,6 +363,18 @@ def test_config(tmp_path: Path) -> None:
             conn.execute(text("create table hello2 (i int)"))
 
 
+def test_url_config(tmp_path: Path) -> None:
+    eng = create_engine("duckdb:///:memory:?worker_threads=123")
```
Comment (Mause) [inline duckdb_engine/tests/test_basic.py:367]: Could you add a test that uses both config_args and query parameters?

**Review by Mause — CHANGES_REQUESTED** [review]

**Comment by elefeint** [comment]
Sorry, I was at a conference last week. Thank you for fixing this PR up!
Is there something else I can do on this?

**Pull request merged**